### ISSUE TITLE
Cannot Load Obsidian Plugin

### ISSUE THREAD

**Issue by sstubbs** [Automattic/harper#338]
I just wanted to say thank you for this amazing project. I am really keen on trying this out. After installing and enabling the plugin I do not get any error highlighting like I think I should be getting. When restarting Obsidian It hangs and says: plugin "harper" is taking long to load. This is on MacOS.

**Comment by elijah-potter** [comment]
That's not great! To figure this out, would you mind doing three things for me?

1. Make sure you're using the latest version, which (at the time of writing) should be `v0.13.0`.
2. Type out `This is an test` in an open file. Do you get any red underlines?
3. Take a screenshot of the problem, including the developer console. You can do this on MacOS by pressing `Cmd`+`Option`+`I` at the same time.

**Comment by mxgrn** [comment]
Not OP, but having the same exact issues.

![Image](https://github.com/user-attachments/assets/3350fb77-ab4f-406c-abab-c95ff6971db2)

**Comment by murtuzaalisurti** [comment]
same for me, on macOS, latest version 0.13.0

![Image](https://github.com/user-attachments/assets/57ae4499-d2a0-4fe2-83b9-c2c9748594f1)

**Comment by elijah-potter** [comment]
Hmmm, I'm not able replicate the issue on my end. @mxgrn would do take a similar screenshot, but swap over to the "Console" tab at the top of the developer tools? We'll get this fixed in a jiffy.

**Comment by mxgrn** [comment]
![Image](https://github.com/user-attachments/assets/570f20c6-4eaa-4872-a668-549ab8c8894e)

... and after scrolling down a page:

![Image](https://github.com/user-attachments/assets/f1743351-8a33-4f51-a7c1-ff4be0a353bf)

**Comment by elijah-potter** [comment]
Alright. I've created a new release of the Obsidian plugin that _might_ fix your issue. Go ahead and update to `v0.14.0`.

If it doesn't immediately start working, try disabling Web Worker usage in settings. At worst, I'll get a better understanding of the issue. I'm still unable to replicate it on my end (on MacOS or any other platform), so thank you for all information you've been able to give me.

**Comment by mxgrn** [comment]
First, a dev console screenshot
![Image](https://github.com/user-attachments/assets/dc0aa4fd-1a50-47a1-a4de-92cd5ac7f3df)

**Comment by mxgrn** [comment]
Not seeing Harper in Obsidian settings:

![Image](https://github.com/user-attachments/assets/98f48132-5768-4e5f-a44f-310065368162)

Maybe due to:

![Image](https://github.com/user-attachments/assets/7ef964b1-dab9-4ddb-bc24-c907896b1fb4)

**Comment by murtuzaalisurti** [comment]
@elijah-potter worked for me on a new installation of obsidian on a new machine

Obsidian v1.7.7
Harper v0.14.0

![Image](https://github.com/user-attachments/assets/63fbf4b3-be39-4188-8708-5469802e6b5d)

**Comment by murtuzaalisurti** [comment]
unfortunately, it doesn't work on macOS.

**Comment by brisu** [comment]
Experiencing a similar issue. The plugin doesn't block startup but will return the same TypeError when I try to activate the plugin manually. 

OS: Mac OS Sequoia 15.2 (24C101) Apple Silicon 
Obsidian v1.7.7
Harper v0.15.0

Troubleshooting: 

1. Turn off all plugins (also tried creating a brand new vault)
2. Install Harper
3. Enable breakpoints for erros
4. Enabling Harper leads to the following errors `Error: ENOENT: no such file or directory, open '/Users/xxx/Desktop/test obsidian/test/.obsidian/plugins/harper/data.json' at async open (node:internal/original-fs/promises:636:25) at async Object.readFile (node:internal/original-fs/promises:1246:14)`
5. I've tried copying `data.json` from a Windows installation to the indicated directory. It returns the error attached below. 

```
plugin:harper:2278 Switching to `WorkerLinter`
plugin:harper:2402 (3) [e, e, Array(4)]
1b2ac272-4336-4dcf-9ca4-37629ba4ae48:722 Uncaught (in promise) TypeError: WebAssembly.instantiate(): Argument 0 must be a buffer source
    at ft (1b2ac272-4336-4dcf-9ca4-37629ba4ae48:722:33)
    at Object.H [as default] (1b2ac272-4336-4dcf-9ca4-37629ba4ae48:865:44)
    at async k (1b2ac272-4336-4dcf-9ca4-37629ba4ae48:16:10)
    at async et.initialize (1b2ac272-4336-4dcf-9ca4-37629ba4ae48:25:17)
    at async et.setup (1b2ac272-4336-4dcf-9ca4-37629ba4ae48:30:5)
    at async _t (1b2ac272-4336-4dcf-9ca4-37629ba4ae48:118:11)
ft @ 1b2ac272-4336-4dcf-9ca4-37629ba4ae48:722
H @ 1b2ac272-4336-4dcf-9ca4-37629ba4ae48:865
await in H (async)
self.onmessage @ 1b2ac272-4336-4dcf-9ca4-37629ba4ae48:123
```

Would love to have this working on MacOS. Any way to help or look further into this?

**Comment by craigz** [comment]
I've just updated `Harper` to version `0.16.0` and now have the same issue as described in comments above. 

- enabling the plug-in & reloading third-party plug-ins shows `Harper` enabled, but there is no cog icon to access settings.
- `Harper` does not appear in the list of `Community plugins`. 

Restarting `Obsidian` results in the `Plugin "Harper" is taking long to load` screen which persists until I click the `disable` button. Following this, `Obsidian` loads as expected, with `Harper` disabled.

I'm running Obsidian on MacOS 15.2.

**Comment by stephenstubbs** [comment]
The problem still persists for me too.

**Comment by makariostang** [comment]
Hi, I have a similar issue on Windows 11.

Harper v0.16.0
Obsidian v1.7.7

![Image](https://github.com/user-attachments/assets/1a0a910f-4372-44f4-9101-893abbc33707)

**Comment by elijah-potter** [comment]
Alright. I have reason to believe this issue derives from older versions of Electron not including `fetch` dependencies in Web Worker contexts. I'm drafting a fix now.

**Comment by elijah-potter** [comment]
Alright, after a good amount of debugging, I've nailed down exactly what's causing this issue. In older versions of Electron, the Request object isn't available in certain contexts. 

Harper expects an up-to-date version of Electron. If you have this issue, [reinstall Obsidian](https://obsidian.md/download) or otherwise update your Electron version.

**Comment by murtuzaalisurti** [comment]
@elijah-potter re-installed obsidian with the latest installer (1.84), it works now on macOS.

**Comment by elijah-potter** [comment]
@stephenstubbs, can you confirm that it works if you update Electron? Otherwise, I'd like to close this issue.

**Comment by makariostang** [comment]
@elijah-potter Thank you, works for me on windows 11 after reinstall too.

**Comment by elijah-potter** [comment]
Sweet! I'm going to close this issue then. Feel free to reopen if it's still a problem.

**Comment by stephenstubbs** [comment]
@elijah-potter it works! Thank you very much.

**Comment by elijah-potter** [comment]
No, thank you @stephenstubbs. I never would have known this was a problem.
